generic3g/vertical/ModelVerticalGrid.F90

```diff
@@ -65,89 +63,99 @@ module function can_connect_to(this, src, rc)
 
 contains
 
-   function new_ModelVerticalGrid_basic(num_levels, units) result(vgrid)
+   function new_ModelVerticalGrid_basic(standard_name, units, num_levels) result(vgrid)
       type(ModelVerticalGrid) :: vgrid
-      integer, intent(in) :: num_levels
+      character(*), intent(in) :: standard_name
       character(*) , intent(in) :: units
-      !# character(*), intent(in) :: short_name
-      !# character(*), intent(in) :: standard_name
-      !# type(StateRegistry), pointer, intent(in) :: registry
+      integer, intent(in) :: num_levels
 
       call vgrid%set_id()
+      vgrid%standard_name = standard_name
       call vgrid%set_units(units)
       vgrid%num_levels = num_levels
-      !# vgrid%short_name = short_name
-      !# vgrid%standard_name = standard_name
-      !# vgrid%registry => registry
    end function new_ModelVerticalGrid_basic
 
    integer function get_num_levels(this) result(num_levels)
       class(ModelVerticalGrid), intent(in) :: this
       num_levels = this%num_levels
    end function get_num_levels
 
-   subroutine add_variant(this, short_name)
+   subroutine add_short_names(this, edge, center)
       class(ModelVerticalGrid), intent(inout) :: this
-      character(*), intent(in) :: short_name
+      character(*), optional, intent(in) :: edge
+      character(*), optional, intent(in) :: center
 
-      call this%variants%push_back(short_name)
-   end subroutine add_variant
+      if (present(edge)) this%short_name_edge = edge
+      if (present(center)) this%short_name_center = center
+   end subroutine add_short_names
+
+   function get_short_name(this, vertical_dim_spec, rc) result(short_name)
+      character(:), allocatable :: short_name
+      class(ModelVerticalGrid), intent(in) :: this
+      type(VerticalDimSpec), intent(in) :: vertical_dim_spec
+      integer, optional :: rc
+
+      if (vertical_dim_spec == VERTICAL_DIM_EDGE) then
+         short_name = this%short_name_edge
+         _RETURN(_SUCCESS)
+      else if (vertical_dim_spec == VERTICAL_DIM_CENTER) then
+         short_name = this%short_name_center
+         _RETURN(_SUCCESS)
+      else
+         _FAIL("unsupported vertical_dim_spec")
+      end if
```

Comment:
War on nesting ...   (Makes it more obvious that a final `_RETURN(_SUCCESS)` is not needed also)
```suggestion
      if (vertical_dim_spec == VERTICAL_DIM_EDGE) then
         short_name = this%short_name_edge
         _RETURN(_SUCCESS)
      end if
      if (vertical_dim_spec == VERTICAL_DIM_CENTER) then
         short_name = this%short_name_center
         _RETURN(_SUCCESS)
      end if
      _FAIL("unsupported vertical_dim_spec")
```

Reply:
There is no nesting here. Also I don't see how separate if blocks is any more readable than the if/else if/else block.

Reply:
I drank the proverbial cool-aid provided by some YouTube videos on "never nesting".   (Relevant text post here: https://dev.to/dglsparsons/write-better-code-and-be-a-better-programmer-by-never-using-else-statements-4dbl)

I will grant that this form is nearly like a SELECT CASE construct which is acceptable by that standard.

I strongly suspect that if we refactor just a bit further we get something that we both agree is more readable.    In any event this is not an example worth spending more time on.  I yield.